Source/Core/Duality/Utility/Coroutines/WaitUntil.cs

```diff
@@ -13,16 +13,18 @@ private enum WaitType
 		{
 			Frames,
 			GameTime,
-			RealTime
+			RealTime,
+			Coroutine
 		}
 
 		/// <summary>
 		/// Waits until the next frame
 		/// </summary>
 		public static readonly WaitUntil NextFrame = new WaitUntil(1, WaitType.Frames);
 
-		private readonly WaitType type;
 		private float internalValue;
+		private readonly Func<bool> subroutineMoveNext;
```

Comment:
To generalize this more make this return a int and call it updaterfunction so we can use for other things as well.